cdap-app-fabric/src/main/java/io/cdap/cdap/security/auth/MessagingAuditLogWriter.java

```diff
@@ -81,37 +82,33 @@ public MessagingAuditLogWriter(MessagingService messagingService,
   }
 
   /**
-   * pushes the collection of log entry to respective messaging topic
-   *
-   * @param auditLogContexts
+   * pushes a log entry to respective messaging topic
    */
   @Override
-  public void publish(@Nullable Queue<AuditLogContext> auditLogContexts) throws IOException {
+  public void publish(@Nullable AuditLogRequest auditLogRequest) throws IOException {
 
-    if (auditLogContexts != null && auditLogContexts.isEmpty()){
+    if (auditLogRequest != null && auditLogRequest.getAuditLogContextQueue().isEmpty()
+      && auditLogRequest.getAuditLogContextQueue().stream().noneMatch(AuditLogContext::isAuditLoggingRequired)){
```

Comment:
Do you mean `||`? Currently it checks for noneMatch in already checked empty collection

Reply:
I would also probably filter out the ones that `!isAuditLoggingRequired` much earlier. Otherwise if you have a mix, you would serialize all of those